### PR TITLE
refactor: prefer prefix ++ operators for non-primitive types (part 2)

### PR DESCRIPTION
**Pull request by rodolforg**
And prefer the style of the `for` loop as discussed in the forums https://forums.synfig.org/t/trying-to-define-our-code-style/13097.
```c++
for (int a = 0; a < 30; ++a) {
}
```
(spaces surrounding operators, spaces before `(` and after and `;`, and bracket layout)

### CODE REVIEW

**Comment by rodolforg** [comment]
Maybe it should be perf, not refactor... lol

**Comment by Keyikedalube** [comment]
Do you want help creating the coding guideline doc page on synfig dev site? I can do that in the evening today if you want :)

File: synfig-core/src/synfig/filecontainerzip.cpp [hunk 1]
```diff
@@ -343,8 +343,7 @@ void FileContainerZip::read_history(std::list<HistoryRecord> &list, FILE *f, fil
 	bool found = false;
 	HistoryRecord history_record;
 
-	for(int i = read_size - sizeof(EndOfCentralDirectory); i >= 0; i--)
-	{
+	for (int i = read_size - sizeof(EndOfCentralDirectory); i >= 0; i--) {
```
Comment (Keyikedalube) [inline synfig-core/src/synfig/filecontainerzip.cpp:346]: Postfix seems fine to me logically unless the rh value is needed immediately.

Below you changed all to prefixes. Here I guess you missed the `i++` :P

**Review by Keyikedalube — APPROVED** [review]

File: synfig-core/src/synfig/filecontainerzip.cpp (hunk 1 above)
Comment (rodolforg) [reply synfig-core/src/synfig/filecontainerzip.cpp:346]: Some "good practices" guides recommend always prefer pre-increment operator in `for` loops. The reason is to not make mistakes and not forget to use it when is not a primitive type.

Comment (Keyikedalube) [reply synfig-core/src/synfig/filecontainerzip.cpp:346]: I saw some Internet post about pre increment for user data types saving memory. Never realized post might make non-primitive inefficient dude to temp obj creation.

Your PR was kinda a good CS revision for me LOL
Glad I reviewed and learned something along the way 😅

**Pull request merged**